home-manager/home-manager

```diff
@@ -165,7 +165,9 @@ function doEdit() {
 
     setConfigFile
 
-    exec "$EDITOR" "$HOME_MANAGER_CONFIG"
+    # See https://github.com/nix-community/home-manager/issues/1496 as to why
+    # `eval` is preferable to `exec`.
+    eval -- "$EDITOR" "$HOME_MANAGER_CONFIG"
```

Comment:
Hmm, would this work if `$EDITOR` contains a space in the path to the executable? I.e., something like
```shell
EDITOR="/path to my/editor"
```
I guess this is a more rare case than wanting to put an argument so might be OK to drop support for space in the path.

Also, any particular reason why removing the quotes around editor doesn't work? I.e.
```shell
exec $EDITOR "$HOME_MANAGER_CONFIG"`
```

Reply:
> I guess this is a more rare case than wanting to put an argument so might be OK to drop support for space in the path.

Yeah this prob won't work, but I can't imagine that this is a common use case to begin with. Spaces in paths are always problematic.

> Also, any particular reason why removing the quotes around editor doesn't work? I.e. `exec $EDITOR "$HOME_MANAGER_CONFIG"`

This does seem to work as well:
```
bash -c 'exec code --wait "/home/skainswo/.ssh/config"'
```
works but
```
bash -c 'exec code --wait "~/.ssh/config"'
```
does not. But I don't think that's an issue if we're already using `exec` anyhow. Happy to do either `eval` or `exec`!

Reply:
I slightly prefer the exec variant since it won't leave an unnecessary home-manager process hanging around. But I imagine it's not a big difference regardless. Let me know what you prefer and I'll merge afterwards.

Reply:
Ok, just pushed 049a279 using the `exec` variant.